src/Symfony/Component/Console/Application.php

```diff
@@ -108,7 +108,7 @@ public function setCommandLoader(CommandLoaderInterface $commandLoader)
         $this->commandLoader = $commandLoader;
     }
 
-    public function getSignalRegistry(): SignalRegistry
+    public function getSignalRegistry(): ?SignalRegistry
```

Comment:
I don't like this approach. It's a bit poor in term of DX.
People will have to check if the SignalRegistry is defined before using it.
And if it's not defined, they don't know why it's not defined. They have to read the Symfony Code to understand what's it's going.

Instead, I would revert this change and I would throw an exception, like this one

https://github.com/symfony/symfony/blob/cbfa540b02394ce103f6235b58296cfda22dea07/src/Symfony/Component/Console/Application.php#L958

Reply:
👍  should be better now